sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/Status.kt

```diff
@@ -27,8 +28,8 @@ sealed class Status(val originatingElement: ParserRuleContext) {
         dependencies: Collection<Any>
     ) : ValidationStatus(element, dependencies)
     class Invalid(
-        val exceptions: Collection<SqlitePluginException>,
+        val errors: Collection<ResolutionError>,
         dependencies: Collection<Any>
-    ) : ValidationStatus(exceptions.first().originatingElement, dependencies)
+    ) : ValidationStatus(errors.first().originatingElement, dependencies)
```

Comment:
The fact that we grab the element from the first error seems like a modeling problem. Either each error has a distinct element or the validation happens on one element.

We can address this in a follow up. It was one of the first things I tried to tackle when I was making attempts at refactoring this stuff as well.